lib/buffer.js

```diff
@@ -348,6 +348,21 @@ Buffer.concat = function(list, length) {
   return buffer;
 };
 
+Buffer.encode = function(source, encoding, targetEncoding) {
+  if (typeof source !== 'string')
+    throw new TypeError('"source" argument must be a string');
+  if (!targetEncoding) {
+    targetEncoding = encoding;
+    encoding = 'utf8';
+  }
+  targetEncoding = targetEncoding || 'utf8';
+  encoding = encoding || 'utf8';
+  if (source === '' || encoding === targetEncoding)
+    return source;
+  if (encoding === 'hex' && source.length % 2 !== 0)
```

Comment:
It probably doesn't matter much.  I'm fairly sure even the baseline compiler optimizes it to a bit test.